src/core/mac/sub_mac.cpp

```diff
@@ -261,14 +264,22 @@ void SubMac::HandleReceiveDone(RxFrame *aFrame, Error aError)
         UpdateFrameCounter(aFrame->mInfo.mRxInfo.mAckFrameCounter);
     }
 
-#if OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE
+#if OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE || OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
     if (aFrame != nullptr && aError == kErrorNone)
     {
+        // Assuming the risk of the parent missing the Enh-ACK in favor of smaller CSL receive window
+        if ((mCslPeriod > 0) && aFrame->mInfo.mRxInfo.mAckedWithSecEnhAck)
+        {
+            mCslLastSync = TimeMicro(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance())));
```

Comment:
With ±20 ppm crystals in both parent and child, a 100 us error here has an impact of 8 us in the receive window duration. Do you think it's worth adding more code here to address it?

Reply:
Not necessarily in this PR. However the problem we should address is undeterministic latency of execution from the main thread. This latency causes that timestamps or transmitted packets (captured here) may be incorrect. We address such undeterministic latency in other places with kCslTxTimeAhead, RxTimeAhead, etc.

We should think of a way to address this problem here as well. I think the cleanest solution would be to get precise TX timstamps from the 802.15.4 driver, but it does not expose such feature yet. Temporarily we could use another safety constant like kCslTxTimestampLatency and its value could be selected for given application depending on expected latencies in the main thread execution.

Reply:
I changed it to use `mTimestamp`. For the RX path we should be on the safe side since the timestamp is prior to the sync Enh-ACK. For the TX path we get the timestamp of the received ACK, which is later than the sync TX frame, but always bounded.

Reply:
I agree. I've already commented: https://github.com/openthread/openthread/pull/6311#discussion_r599713984